lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -502,6 +502,38 @@ static LogicalResult verifyInstanceOp(InstanceOp &instance) {
     return failure();
   }
 
+  // Check that the result type is consistent with its module.
+  if (auto referencedFModule = dyn_cast<FModuleOp>(referencedModule)) {
+    auto bundle = instance.getResult()
+                      .getType()
+                      .cast<FIRRTLType>()
+                      .getPassiveType()
+                      .cast<BundleType>();
+    auto bundleElements = bundle.getElements();
+    size_t e = bundleElements.size();
+
+    if (e != referencedFModule.getNumArguments()) {
+      instance.emitOpError()
+          << "has a wrong size of bundle type, expected size is "
+          << referencedFModule.getNumArguments() << " but got " << e;
+      return failure();
+    }
+
+    for (size_t i = 0; i != e; i++) {
+      auto expectedType = referencedFModule.getArgument(i)
+                              .getType()
+                              .cast<FIRRTLType>()
+                              .getPassiveType();
+      if (bundleElements[i].second != expectedType) {
+        instance.emitOpError("output bundle type must match module. In "
+                             "element ")
+            << i << ", expected " << expectedType << ", but got "
+            << bundleElements[i].second;
```

Comment:
I'd add a reference to the original module declaration. Like the suggestion above.

Nit: I'd add a period to the end of this message since it's being broken up into two sentences. @lattner may have a better policy suggestion here?

Reply:
I added a period for now. Thanks